gems/sorbet-runtime/lib/types/_types.rb

```diff
@@ -126,7 +126,7 @@ def self.type_parameter(name)
   # exception at runtime if the value doesn't match the type.
   #
   # Compared to `T.let`, `T.cast` is _trusted_ by static system.
-  def self.cast(value, type, checked: true)
+  def self.cast(value, type, checked: T::Configuration.runtime_type_assertions_enabled?)
```

Comment:
Noting before I forget: this will incur the runtime penalty of calling `T::Configuration.runtime_type_assertions_enabled?` every time `T.let` / `T.cast` / etc. is called.

I wonder if there are alternatives that don't suffer from this.

Reply:
Good shout. I'll give it some thought!

Reply:
I forgot to get to this the weekend but I had a thought:

Similar to how setting the `default_checked_level` to `:never`, `:tests`, or `:always` gives you one chance to call it, could we do a similar thing with this kinda like:

- User calls something like `runtime_type_assertions(:disable)`, presumably really early in the application booting.
- We do like `T.singleton_class.prepend` new versions of the methods that are functionally noops
- That's it it's done and no longer configurable.

It should avoid any runtime penalty of checking another method call and it follows an existing pattern for setting something scary. 
Is that sensible? Again keen for feedback.

Reply:
> * We do like `T.singleton_class.prepend` new versions of the methods that are functionally noops

Heads up that we do not (cannot) use `prepend` in sorbet-runtime. Stripe's codebase does not support it. You can see previous discussions around not using `prepend` here: https://github.com/sorbet/sorbet/pull/2078

I think it would be fine to remove the original method and replace it with another method (we already have a helper for this, it's how sigs work. Grep for `replace_method`.

If you want to try implementing a solution that uses `replace_method` plus the "one-shot" logic you mention above, I'd be open to looking at the code.

(and again: please take a look at my question of motivation below, because if this is for primarily performance concerns, this will not solve them)